index.js

```diff
@@ -228,10 +235,30 @@ module.exports = function (schema, options) {
         });
     }
 
-    schema.methods.delete = function (deletedBy, cb) {
-        if (typeof deletedBy === 'function') {
-          cb = deletedBy;
-          deletedBy = null;
+    schema.methods.delete = function (params, cb) {
+        let deletedBy = null;
+        let deletedId = null;
+
+        if (typeof params === 'function') {
+          cb = params;
+          params = {};
+        }
+
+        // If options is a string or ObjectId, it's the deletedBy value
+        if (typeof params === 'string' || typeof params === 'object' && !params.deletedId ||  (params && params._bsontype === 'ObjectID')) {
+            deletedBy = params;
+            params = {};
+        } else if (!params) {
+            params = {};
+        }
```

Comment:
🤏 If you set the `params` fallback in the prototype then you should be able to drop this
```suggestion
        }
```